osu.Game/OsuGame.cs

```diff
@@ -221,13 +221,18 @@ void setBeatmap()
                     return;
                 }
 
-                var databasedSet = BeatmapManager.QueryBeatmapSet(s => s.OnlineBeatmapSetID == beatmap.OnlineBeatmapSetID);
+                var databasedSet =
+                    BeatmapManager.QueryBeatmapSet(s => s.OnlineBeatmapSetID == beatmap.OnlineBeatmapSetID) ??
```

Comment:
can also write as `BeatmapManager.QueryBeatmapSet(s => s.OnlineBeatmapSetID == beatmap.OnlineBeatmapSetID || s.Hash == beatmap.Hash)`. this would be more optimal (less database queries) but i'm not sure which one is more logically correct.